docker-entrypoint.sh

```diff
@@ -292,6 +292,31 @@ docker_setup_db() {
 		EOSQL
 	fi
 
+	local mysqlAtLocalhost=
+	local mysqlAtLocalhostGrants=
+	# Install mysql@localhost user
+	if [ -n "$MARIADB_MYSQL_LOCALHOST_USER" ] || [ -n "$MARIADB_MYSQL_LOCALHOST_GRANTS" ]; then
+		local pw=
+		pw="$(pwgen --numerals --capitalize --symbols --remove-chars="'\\" -1 32)"
+		# MDEV-24111 before MariaDB-10.4 cannot create unix_socket user directly auth with simple_password_check
+		# It wasn't until 10.4 that the unix_socket auth was built in to the server.
+		read -r -d '' mysqlAtLocalhost <<-EOSQL || true
+		EXECUTE IMMEDIATE IF(VERSION() RLIKE '^10\.[23]\.',
+			"INSTALL PLUGIN /*M10401 IF NOT EXISTS */ unix_socket SONAME 'auth_socket'",
+			"SELECT 'already there'");
+		CREATE USER mysql@localhost IDENTIFIED BY '$pw';
+		ALTER USER mysql@localhost IDENTIFIED VIA unix_socket;
+		EOSQL
+		if [ -n "$MARIADB_MYSQL_LOCALHOST_GRANTS" ]; then
+			if [[ "$MARIADB_MYSQL_LOCALHOST_GRANTS" = ALL* ]] || \
```

Comment:
Are we really confident with this list, i.e. are these the only ones that might *ever* cause *any* issues? A priv escalation is not just about creating new users, but also about doing any stuff that a user isn't supposed to - including DoS'ing the instance. I'm thinking about `ALTER`, `DROP`, `DELETE`. But also what about `FILE` or `SHUTDOWN`...

Instead of providing a list of privileges that might be bad (i.e. a blocklist approach) we should rather always show the warning except when we know that they are safe (i.e. a allowlist). And the only privilege we're actually sure about is `USAGE`. So, why not:

```sh
if [ "$MARIADB_MYSQL_LOCALHOST_GRANTS" != "USAGE" ]; then
```

Reply:
You're right, every privilege exists because it can do something extra ordinary, so sure, I'll commit the `!= USAGE ` approach.